articles/modules/ROOT/pages/how-to-force-match-ordering.adoc

```diff
@@ -0,0 +1,107 @@
+
+Keep in mind also that Where clauses are not independent, and are bound to the preceding Match or With clause.
+This means that since multiple Match clauses are evaluated during planning, their Where clauses are likewise available for consideration.
+
+It is also important to understand that a With clause alone also cannot prevent the planner from considering Match patterns after it.
+
+This article discusses the reasons for this behavior, offers examples of when this behavior can become problematic, and provides techniques you can use to force the planner to fulfill separate Match clauses in the order they occur in your query.
+
+== In a query plan there is no Match, there is only Expand
+
+While the Cypher query describes "what" you want to find, the execution plan operators that make up a query plan are "how" Neo4j will fulfill the query.
+
+If you review https://neo4j.com/docs/cypher-manual/current/execution-plans/operator-summary/[query plan operators in the docs], you may note that they do not correspond one-to-one with Cypher clauses.
+There is no Match operator.
+Instead, to fulfill anchoring and expansion, there are several kinds of lookup operators (AllNodesScan, NodeByLabelScan, NodeIndexSeek, and others), and several kinds of expand operators.
+
+As such, this particular aspect of query planning involves the analysis of the Match patterns in a query (not Match pattern by Match pattern, but connected patterns across multiple Matches),
+and breaking those down into these various smaller operations.
+The ordering of those smaller operations is not constrained by the ordering of the Match patterns that these operations fulfill.
+Instead, the planner will attempt to use what it knows of the graph via metadata and counts data in order to select an optimal plan.
+
+In some cases, this works well, as the planner is able to see the larger pattern across Matches and consider other options that may be better for lookup and anchoring.
+
+In other cases, the metadata available to the planner may not give it sufficient knowledge to choose an optimal plan, and a suboptimal one may result instead.
+It is also possible for a plan to be mostly optimal across most of the graph data it can match against, but for there to be exceptional cases, such as supernodes with dense relationships, that end up being severely suboptimal when the query executes across these exceptional areas of the graph.
+
+
+== Barriers to entry
+
+We cannot directly dictate the ordering of how Matches are evaluated.
+We can, however, use some Cypher techniques to introduce barriers in a query that the planner cannot cross when planning how to solve a Match clause.
+
+Consider a graph of users, the tv shows they like, and the country each lives in.
+A query for one such path, with all 3 nodes of the path strictly defined, might be this:
```

Comment:
should you mention that only happens if there is an index/constraint on all three? otherwise the one with the index will be picked as starting point.